wip: map view

bradh opened the pull request:
## Motivation and Context
Draft for comment: adds a map view option to the viewer

## Description
Uses the LGPL licensed JXMapViewer2 component, along with some custom rendering.

## How Has This Been Tested?
Viewer with some test data.

## Screenshots (if appropriate):
![image](https://user-images.githubusercontent.com/174642/113507017-5db3f680-958b-11eb-9003-7cb35f7a0fc6.png)

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Would like thoughts on how the UI should be handled - should there be a whole style / options thing, or is a resource file enough? Default sizes and colours? Should the map remain centred on the sensor frame centre? Should it zoom in and out automatically? Should the frame corner numbers only show at a certain zoom level?

wlfgang commented:
Tested it a little locally and this looks excellent, thank you! My thoughts on some of your questions:

- I think a resource file would be okay for any options. If the primary user is still a developer/engineer, the UI doesn't need to get too fancy. IMO keeping the size of the viewer code base small is important so that a developer can easily understand it.
- I wouldn't center or zoom automatically by default- might cause user frustration if they want see something off-screen. I think it would be useful as an option though.
- The current size and colors seem good to me.
- Hiding/showing the corner numbers automatically is a good idea, although as it stands the display doesn't seem too cluttered when you zoom out.

bradh commented:
Updated the implementation. I did start on an option dialog, but it didn't seem that usable.  Implemented the resources thing, which includes ability to set the colours.

Following the FoV is a non-default option on the map view "status bar". 

I haven't done the automatic hiding of corners, although that won't be too hard to add later if we need it.

Update is #261.
